images/scripts/update-golang-image.sh

```diff
@@ -25,12 +25,17 @@ image="${1:-docker.io/library/golang:${go_version}}"
 
 image_digest="$("${script_dir}/get-image-digest.sh" "${image}")"
 
+if [ -z "${image_digest}" ]; then
+  echo "Image digest not available"
+  exit 1
+fi
+
 # shellcheck disable=SC2207
-used_by=($(git grep -l GOLANG_IMAGE= images/{builder,hubble-proto,runtime}))
+used_by=($(git grep -l GOLANG_IMAGE= images/*/Dockerfile))
 
 for i in "${used_by[@]}" ; do
     # golang images with image digest
-    sed "s|GOLANG_IMAGE=docker\.io/library/golang:[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*@.*|GOLANG_IMAGE=${image}@${image_digest}|" "${i}" > "${i}.sedtmp" && mv "${i}.sedtmp" "${i}"
+    sed "s|GOLANG_IMAGE=docker\.io/library/golang:[0-9][0-9]*\.[0-9][0-9]*\(\.[0-9][0-9]*\)\?@.*|GOLANG_IMAGE=${image}@${image_digest}|" "${i}" > "${i}.sedtmp" && mv "${i}.sedtmp" "${i}"
     # other golang images (e.g. golang-alpine images)
-    sed "s|GOLANG_IMAGE=docker\.io/library/golang:[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*-\(.*\)|GOLANG_IMAGE=${image}-\1|" "${i}" > "${i}.sedtmp" && mv "${i}.sedtmp" "${i}"
+    sed "s|GOLANG_IMAGE=docker\.io/library/golang:[0-9][0-9]*\.[0-9][0-9]*\(\.[0-9][0-9]*\)\?-\(.*\)@.*|GOLANG_IMAGE=${image}-\2@${image_digest}|" "${i}" > "${i}.sedtmp" && mv "${i}.sedtmp" "${i}"
```

Comment:
It looks like this change breaks update of `GOLANG_IMAGE` in `images/hubble-proto/Dockerfile`. Updating `GO_VERSION` to `1.16.0` and running `make update-golang-image` updates the Go version everywhere but in `images/hubble-proto/Dockerfile`. Also I think ${image_digest} will be wrong for the `-alpine*` image.